pkg/drivers/kic/oci/network.go

```diff
@@ -82,6 +82,17 @@ func RoutableHostIPFromInside(ociBin string, clusterName string, containerName s
 		return containerGatewayIP(ociBin, containerName)
 	}
 
+	// on mac os and podman, try lima host record
+	if ociBin == "podman" && runtime.GOOS == "darwin" {
```

Comment:
Thank you for this PR, a couple questions

1- why this is only for pod man? Could this approach be used it be Docker on Lima  on macO too?

2- it seems to me this gonna use host.ima.internal for all Podmans on MacOS 
(Including if the user is gonna use Podman-Machine https://github.com/afbjorklund/podman-machine

Could we instead write a detection logic, to check if we are running inside Lima VM ? We have a package called “Detect” that has all kind of logic to detect where were are running (maybe we need a new detect for running inside Lima

https://github.com/kubernetes/minikube/blob/master/pkg/minikube/detect/detect.go


CC: Adding to comment here.
@afbjorklund

Reply:
Hi @medyagh thanks for your time reviewing this,

1- it would be for any lima created VM i think yes. I am not sure on how to detect well at the moment, i will look in the lima code what would be the most sensible way to detect we are running in a lima vm.
2- i don't think  that podman use lima as a vm builder on windows. In fact podman doesn't use lima on macos too, it is only if you build you lima vm for podman and configure podman that lima will be used (else podman-machine use 
straight qemu with gsock wivor tap for network - i prefer to use lima because it support well socket_vmnet which allows to create brdiged network interfaces in the VM)
3 - limavm is not used by default by podman. i agree that i should add some detection logic in the file you mention so whatever the case if are using a limavm we can use the domain name to get the gateway ip, i will look for this.

thanks a lot for your comments i look forward to improve the code in the way described here, have a good day!

Reply:
yes please if this is not something that default users use, we need to only apply it for specific environment that is being used